gnomonicus/gnomonicus_lib.py

```diff
@@ -43,198 +38,117 @@ def __init__(self, gene: str, mutation: str):
-                outdated = True
+                item = [int(v) for v in value]
         else:
-            outdated = True
-        if outdated:
-            logging.info("Genome object is outdated! Trying with the original filepath")
-    except Exception as e:
-        logging.info(
-            f"No pickled version of genome object, instanciating and dumping. Error: {e}"
-        )
-
-    # Create new gumpy.Genome and pickle dump for speed later
-    reference = gumpy.Genome(path, show_progress_bar=progress)
-    reference.gumpy_version = gumpy.__version__
-    pickle.dump(reference, open(path + ".pkl", "wb"))
-    return reference
-
-
-def loadGenomeAndGenes(
-    path: str, progress: bool
-) -> tuple[gumpy.Genome, dict[str, gumpy.Gene]]:
-    """Look for pickled genome and genes, instanciating one or both if required.
-
-    Used by the table-update script
-
-    Args:
-        path (str): Path to the genbank file or pickle dump. If previously run, a genbank file's Genome object is pickled and dumped for speed
-        progress (bool): Boolean as whether to show progress bar for gumpy
-
-    Returns:
-        tuple[gumpy.Genome, dict[str, gumpy.Gene]]: Tuple of (reference genome object, Dictionary mapping gene name -> gumpy.Gene)
-    """
-    reference = loadGenome(path, progress)
-    gene_path = path + ".genes.pkl.gz"
-
-    if checkGzip(gene_path):
-        # Genes already exist so load
-        with gzip.open(gene_path, "rb") as f:
-            genes = pickle.load(f)
-    else:
-        # Genes didn't exist, so build then dump
-        genes = {}
-        for gene_name in tqdm(reference.genes, disable=not progress):
-            genes[gene_name] = reference.build_gene(gene_name)
-        with gzip.open(gene_path, "wb") as f:
-            pickle.dump(genes, f)
-
-    return reference, genes
+            if item is None or isinstance(item, int) or isinstance(item, float):
+                # Should never happen but appease mypy
+                continue
+            for v in value:
+                # Use duck typing to determine if it's a float or int
+                try:
+                    item.append(int(v))
+                except ValueError:
+                    try:
+                        item.append(float(v))
+                    except ValueError:
+                        item.append(v)
+        ev[key] = item
+    for key, val in ev.items():
+        if isinstance(val, list) and len(val) == 1:
+            # Unpack single values as they probably shouldn't be lists
+            ev[key] = val[0]
+    # We also want to add back in some of the VCF items which aren't in the fields dict
+    ev["POS"] = evidence.position
+    ev["REF"] = evidence.reference
+    ev["ALTS"] = evidence.alternative
+    return ev
 
 
 def populateVariants(
     vcfStem: str,
     outputDir: str,
-    diff: gumpy.GenomeDifference,
+    diff: grumpy.GenomeDifference,
     make_csv: bool,
     resistanceGenesOnly: bool,
+    sample: grumpy.Genome,
```

Comment:
Have you added this arg to the docstring?

Reply:
I had not, good catch